crates/starknet-server/tests/test_dump_and_load.rs

```diff
@@ -217,4 +247,171 @@ mod dump_and_load_tests {
+                .spawn()
+                .unwrap();
+            kill.wait().unwrap();
+        }
+
+        // file should not be created if there are no transactions
+        if Path::new(dump_file_name).exists() {
+            panic!(
+                "Could find the dump file but there were no transactions to dump {}",
+                dump_file_name
+            );
+        }
+    }
+
+    #[tokio::test]
+    async fn dump_endpoint_fail_with_wrong_request() {
+        let devnet_dump = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+        let dump_body = Body::from(
+            json!({
+                "test": ""
+            })
+            .to_string(),
+        );
+        let result = devnet_dump.post_json("/dump".into(), dump_body).await.unwrap();
+        assert_eq!(result.status(), 422);
+    }
+
+    #[tokio::test]
+    async fn load_endpoint_fail_with_wrong_request() {
+        let devnet_load = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+        let load_body = Body::from(
+            json!({
+                "test": ""
+            })
+            .to_string(),
+        );
+        let result = devnet_load.post_json("/load".into(), load_body).await.unwrap();
+        assert_eq!(result.status(), 422);
+    }
+
+    #[tokio::test]
+    async fn dump_endpoint_fail_with_wrong_file_name() {
+        let devnet_dump = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+        let dump_body = Body::from(
+            json!({
+                "path": "///"
+            })
+            .to_string(),
+        );
+        let result = devnet_dump.post_json("/dump".into(), dump_body).await.unwrap();
+        assert_eq!(result.status(), 500);
+    }
+
+    #[tokio::test]
+    async fn load_endpoint_fail_with_wrong_path() {
+        let load_file_name = "load_file_name";
+        let devnet_load = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+        let load_body = Body::from(
+            json!({
+                "path": load_file_name
+            })
+            .to_string(),
+        );
+        let result = devnet_load.post_json("/load".into(), load_body).await.unwrap();
+        assert_eq!(result.status(), 400);
+    }
+
+    #[tokio::test]
+    async fn dump_load_endpoints_transaction_and_state_after_load_is_valid() {
+        let dump_file_name = "dump_endpoint";
+        let dump_file_name_custom_path = "dump_endpoint_custom_path";
+        let devnet_dump =
+            BackgroundDevnet::spawn_with_additional_args(&["--dump-path", dump_file_name])
+                .await
+                .expect("Could not start Devnet");
+        let mint_tx_hash = devnet_dump.mint(DUMMY_ADDRESS, DUMMY_AMOUNT).await;
+        let dump_body = Body::from(
+            json!({
+                "path": ""
+            })
+            .to_string(),
+        );
+        devnet_dump.post_json("/dump".into(), dump_body).await.unwrap();
```

Comment:
I don't understand, what is the purpose of this second dumping in this test? We aren't loading it anywhere, right? Should it be a separate case? Again, I might be wrong because I don't understand the intended purpose.

Reply:
The first dump should take the default path which is "dump_endpoint" and we asserting that. Secondly, we check the second dump with a custom path "dump_endpoint_custom_path".

Reply:
I'll add comments with that

Reply:
Added comments here 63e664ee8429af41b9ca690d52682f0f2dae2f31

Reply:
But why are there two dumps and only one load? Also, is there a typo in this comment (is -> if):
```
load and re-execute from "dump_endpoint" file and check is transaction
```

Reply:
I'm testing functionality if /dump endpoint with an empty JSON body works and the default path is taken in that case. Later I'm testing if I can still set the custom path and dump to the custom path and the path will be different (not default). Files are the same so I could load any of them later.

Typo fixed in 0e1bd79aa51cd93ea82d372e18a1256cae10c6ee